ping_manager/ping_manager.py

```diff
@@ -0,0 +1,232 @@
+            {"$pull": {"ignored_categories": category_to_ignore.id}},
+        )
+        await ctx.send(f":+1: Removed {category_to_ignore} from the ignored categories list.")
+
+    async def should_ping(self, channel: discord.TextChannel) -> bool:
+        """Check if a ping should be sent to a thread depending on current config."""
+        if channel.category_id in self.config.ignored_categories:
+            log.info("Not pinging in %s as it's currently in an ignored category", channel)
+            return False
+
+        first_message_list = await channel.history(limit=1, oldest_first=True).flatten()
+        first_message = first_message_list[0]
+        thread_author_name = first_message.embeds[0].author.name
+        async for message in channel.history(oldest_first=True):
+            if not message.embeds:
+                log.info("Not pinging in %s as a mod has sent an internal message in the thread.", channel)
+                return False
+            if not message.embeds[0].author.name == thread_author_name:
+                log.info("Not pinging in %s as a mod has sent a reply in the thread.", channel)
+                return False
+        return True
+
+    async def maybe_ping_later(self, ping_task: PingTask) -> None:
+        """Pings conditionally after waiting the configured wait duration."""
+        when_to_ping = datetime.fromisoformat(ping_task.when_to_close)
+        now = datetime.utcnow()
+        seconds_to_sleep = (when_to_ping - now).total_seconds()
+        if seconds_to_sleep < 0:
+            log.info("Pinging for %d is overdue, pinging now.", ping_task.channel_id)
+        else:
+            await asyncio.sleep(seconds_to_sleep)
+
+        if not (channel := self.bot.get_channel(ping_task.channel_id)):
+            log.info("Channel closed before we could ping.")
+        else:
+            with contextlib.suppress(discord.NotFound):
+                # Ensure that if the channel gets deleted during processing, the task still gets removed.
+                if await self.should_ping(channel):
+                    await channel.send(self.config.ping_string)
+
+        self.ping_tasks.remove(ping_task)
+        await self.db.find_one_and_update(
+            {"_id": "ping-delay-tasks"},
+            {"$pull": {"ping_tasks": asdict(ping_task)}},
+        )
+
+    @commands.Cog.listener()
+    async def on_thread_ready(self, thread: Thread, *args) -> None:
+        """Schedule a task to check if the bot should ping in the thread after the defined wait duration."""
+        await self.init_task
+        now = datetime.utcnow()
+        ping_task = PingTask(
+            when_to_close=(now + timedelta(seconds=self.config.wait_duration)).isoformat(),
+            channel_id=thread.channel.id
+        )
+        self.ping_tasks.append(ping_task)
+        await self.db.find_one_and_update(
+            {"_id": "ping-delay-tasks"},
+            {"$addToSet": {"ping_tasks": asdict(ping_task)}},
+        )
```

Comment:
Thinking more on it, I'm not sure it's worth the time investment, as far as I can tell it only really protects us against the case where connection the the mongo db fails and the bot reboots within 5 minutes, and in that case it would just lead to the ping not going through, which isn't the end of the world.